### PR TITLE
Add python constraint

### LINKED ISSUE #23: Shipping py3-requiring code to py2.7
Issue: New build includes some py3-only code, and is shipping to 2.7

```bash
conda create -n broken-nose -c conda-forge python=2.7 nose
conda activate broken-nose  # see what I did there? 😄 
python -c "import nose" 
```

```pytb
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/__init__.py", line 1, in <module>
    from nose.core import collector, main, run, run_exit, runmodule
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/core.py", line 11, in <module>
    from nose.config import Config, all_config_files
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/config.py", line 6, in <module>
    import configparser
ImportError: No module named configparser
```

(also, that was odd, because I got a different one in my unit tests... so....)

```bash
conda install configparser
python -c "import nose"
```

```pytb
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/__init__.py", line 3, in <module>
    from nose.exc import SkipTest, DeprecatedTest
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/exc.py", line 8, in <module>
    from nose.plugins.skip import SkipTest
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/plugins/skip.py", line 9, in <module>
    from nose.plugins.errorclass import ErrorClass, ErrorClassPlugin
  File "/Users/dwhs/miniconda3/envs/broken-nose/lib/python2.7/site-packages/nose/plugins/errorclass.py", line 130
    class ErrorClassPlugin(Plugin, metaclass=MetaErrorClass):
                                            ^
SyntaxError: invalid syntax
```

(that's what I saw on Travis)

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/dwhs/miniconda3/envs/broken-nose:
#
# Name                    Version                   Build  Channel
backports                 1.0                        py_2    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
certifi                   2019.11.28       py27h8c360ce_1    conda-forge
configparser              3.7.3            py27h8c360ce_2    conda-forge
libcxx                    10.0.1               h5f48129_0    conda-forge
libffi                    3.2.1             hb1e8313_1007    conda-forge
ncurses                   6.2                  hb1e8313_1    conda-forge
nose                      1.3.7           pyh9f0ad1d_1005    conda-forge
openssl                   1.1.1h               haf1e3a3_0    conda-forge
pip                       20.1.1             pyh9f0ad1d_0    conda-forge
python                    2.7.15          h8e446fc_1011_cpython    conda-forge
python_abi                2.7                     1_cp27m    conda-forge
readline                  8.0                  h0678c8f_2    conda-forge
setuptools                44.0.0                   py27_0    conda-forge
sqlite                    3.33.0               h960bd1c_1    conda-forge
tk                        8.6.10               hb0a8c7a_1    conda-forge
wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
zlib                      1.2.11            h7795811_1009    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : broken-nose
    active env location : /Users/dwhs/miniconda3/envs/broken-nose
            shell level : 3
       user config file : /Users/dwhs/.condarc
 populated config files : /Users/dwhs/.condarc
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.7.3.final.0
       virtual packages : __osx=10.14.6
       base environment : /Users/dwhs/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/omnia/osx-64
                          https://conda.anaconda.org/omnia/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/dwhs/miniconda3/pkgs
                          /Users/dwhs/.conda/pkgs
       envs directories : /Users/dwhs/miniconda3/envs
                          /Users/dwhs/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.22.0 CPython/3.7.3 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

(but issue occurred on fresh miniconda on travis)
</details>

### PR DESCRIPTION
**Pull request by dwhswenson**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [N/A] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)  (done 24 hours ago)
* [N/A] Ensured the license file is being packaged. (assume the package author did)

Fixes #23.

Wheels for Python 2/3 contained different source code, so the single (Python 3) tarball doesn't work for Python 2.

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
This PR should add a constraint on python.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,18 +11,17 @@ source:
   sha256: f1bffef9cbc82628f6e7d7b40d7e255aefaa1adb6a1b1d26c69a8b79e6208a98
 
 build:
-  number: 1005
-  noarch: python
+  number: 1006
   entry_points:
```
Comment (beckermr) [inline recipe/meta.yaml:16]:
```suggestion
  noarch: python
  entry_points:
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by jakirkham** [comment]
> This PR should add a constraint on python.

I'm not sure. It uses 2to3, which we typically don't make `noarch`. Admittedly we don't care about building the Python 2 part any more. Though I'm not sure how 2to3 behaves on different Python 3 versions. Since it is part of Python, it is possible that the 2to3 code changes in noticeable ways between Python minor versions.

**Comment by beckermr** [comment]
cc @mariusvniekerk any thoughts here?

**Review by isuruf — APPROVED** [review]

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
@conda-forge/nose ready to merge!

**Comment by moorepants** [comment]
I ask, because I'm not getting build errors in a downstream package that tests against Python 2.7 still using nose. I'm guessing that a Python 3 only compatible version is being installed in my Python 2 conda environment.

**Comment by jakirkham** [comment]
Please see issue ( https://github.com/conda-forge/nose-feedstock/issues/23 )

**Comment by moorepants** [comment]
That's the same error I'm now getting in my builds. So if I need nose in a Py 2.7 environment, how do I get that? I can't say `nose<1.3.7` because the version hasn't changed in 5 years. Is there a prior build number that works with p27? Can I use that somehow?

**Comment by jakirkham** [comment]
Can you please file a new issue sharing what is being run and what the output is? We made a number of changes in the past few days and it's not clear how this issue is being run into since those changes.

**Comment by moorepants** [comment]
Sure. Thanks.

**Comment by moorepants** [comment]
I just reran my CI build where I was getting the failure and no longer do, so maybe I was pulling in nose prior to these new changes before. If anything else comes up I'll make an issue. Good for now.

**Comment by jakirkham** [comment]
Thanks for the update. Glad things are working again 🙂

**Pull request merged**